### PR TITLE
Migrate from Dispatch to OkHttp to prevent dependency clashes with client consumers

### PR DESCRIPTION
**Pull request by annebyrne**

### CODE REVIEW

**Comment by tomrf1** [comment]
Some things to note:

1. okhttp does "transparent" gzip compression, so you don't have to add the gzip header and the response doesn't include it either. See https://github.com/square/okhttp/issues/2132. I verified that it really is coming back compressed from capi by logging `response.networkResponse.headers` (because I'm really paranoid about compression these days).
2. An exception is thrown in libthrift if you don't depend on slf4j-api.

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,6 @@
+## 11.33
+* Move to using OkHTTP over Dispatch
+
```
Comment (mchv) [inline CHANGELOG.md]: We need to keep `model` and `scala client` in sync, so we should release a model change before releasing this.

**Comment by mchv** [comment]
@tomrf1 
> An exception is thrown in libthrift if you don't depend on slf4j-api.

Could you share more on this, I fail to understand how this is possible :scream:

File: CHANGELOG.md (hunk 1 above)
Comment (mchv) [reply CHANGELOG.md]: If anyone has in mind a versioning version that is both simple to understand for consumer and allow to release easily without such constraint, I will be interested to read you!

Comment (regiskuckaertz) [reply CHANGELOG.md]: @mchv move the scala client and the model into two projects within the same repo?

**Comment by regiskuckaertz** [comment]
@annebyrne what about zstandard compression?

File: CHANGELOG.md (hunk 1 above)
Comment (annebyrne) [reply CHANGELOG.md]: I like Regis' suggestion!

**Comment by annebyrne** [comment]
@regiskuckaertz I think it has always just been gzip compression we've had enabled on this client. I don't think its possible for `OkHttp`, who has gzip by default. @mchv might know more about this than me.

**Comment by annebyrne** [comment]
@mchv We are also not sure how this is possible!! I love finding mind-boggling bugs from simple editorial requirements it looks like 😱  :joy:

**Comment by annebyrne** [comment]
Yes I think we can't do zstandard, we use brotli here outside of EdTools where we use zstandard. Don't think OkHttp supports brotli yet (tho maybe @mchv wants to hack around with his filter some more 🤓 )

**Comment by annebyrne** [comment]
Have updated with a consistent versioning across the client and the model. Is this good to go? 🚢 🚀  @mchv @tomrf1 @regiskuckaertz

**Comment by mchv** [comment]
🚢 🚀 💯

**Comment by tomrf1** [comment]
👍 go for it!

**Pull request merged**